lib/index.js

```diff
@@ -177,6 +173,27 @@ exports.request = function (method, url, options, callback, _trace) {
         req.write(options.payload);
     }
 
+    // Custom abort method to detect early aborts
+
+    var _abort = req.abort;
+    var aborted = false;
+    req.abort = function () {
```

Comment:
not a fan over changing the node `req` object.  Can you find another way to solve this?